### PR TITLE
fuzz: remove fuzz cases for deleted code

### PR DESCRIPTION
**Pull request by williambanfield**
This PR removes updates the fuzzer Makefile and github action to reflect the updated directory structure of the fuzz tests.

### CODE REVIEW

File: .github/workflows/fuzz-nightly.yml [hunk 1]
```diff
@@ -23,24 +23,9 @@ jobs:
...
-        continue-on-error: true
-
-      - name: Fuzz p2p-addrbook
-        working-directory: test/fuzz
-        run: timeout -s SIGINT --preserve-status 10m make fuzz-p2p-addrbook
-        continue-on-error: true
-
-      - name: Fuzz p2p-pex
-        working-directory: test/fuzz
-        run: timeout -s SIGINT --preserve-status 10m make fuzz-p2p-pex
```
Comment (cmwaters) [inline .github/workflows/fuzz-nightly.yml:43]: We still have a pex reactor, has the behavior of the pex reactor changed such that these fuzzy tests are no longer relevant?

Comment (williambanfield) [reply .github/workflows/fuzz-nightly.yml:43]: I see that the fuzz tests don't exist anymore. Let me see if I can resurrect the test.

Comment (tychoish) [reply .github/workflows/fuzz-nightly.yml:43]: it's possible that this interacted with the legacy implementation (I did a bunch of deletion here, and I don't remember the exact details, but I definitely deleted some fuzz tests related to the legacy p2p stack. 

Fuzzing the legacy PEX might have been interesting (it was responsible for dialing) but the new PEX is just a front end to the peer database/store and I think wouldn't be super interesting to fuzz

**Review by tychoish — APPROVED** [review]

File: .github/workflows/fuzz-nightly.yml (hunk 1 above)
Comment (williambanfield) [reply .github/workflows/fuzz-nightly.yml:43]: The primary function of the fuzz test that was removed was to check decoding bytes into the pex messages. This is no longer handled in the Pex reactor itself, so the fuzz test no longer really makes sense to maintain.

**Review by creachadair — APPROVED** [review]

**Pull request merged**